### PR TITLE
OHRM5X-2197: Fix claim - related issues

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue [hunk 1]
```diff
@@ -188,6 +184,13 @@ export default {
       }
       return false;
     },
+    computedEmployeeName() {
+      return this.employee.terminationId
+        ? `${this.employee.firstName} ${this.employee.lastName} (${this.$t(
+            'maintenance.past_employee',
+          )})`
+        : `${this.employee.firstName} ${this.employee.lastName}`;
```
Comment (ManulMax) [inline src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue:192]:
```suggestion
? `${this.employee.firstName} ${this.employee.lastName} 
              (${this.$t('maintenance.past_employee')})`
        : `${this.employee.firstName} ${this.employee.lastName}`;
```

**Review by ManulMax — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/pages/SaveClaimEvent.vue [hunk 2]
```diff
@@ -112,15 +112,14 @@ export default {
...
-    //TODO: Refctor this code to use validation API
     this.isLoading = true;
     this.http
       .getAll({limit: 0})
       .then((response) => {
         const {data} = response.data;
         this.rules.name.push((v) => {
           const index = data.findIndex(
-            (item) => item.name.toLowerCase() == v.toLowerCase(),
+            (item) => item.name.toLowerCase() == v.trim().toLowerCase(),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/SaveClaimEvent.vue:122]: @ManulMax check this. need to validate uniqueness in BE validation

File: src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue [hunk 3]
```diff
@@ -188,6 +184,12 @@ export default {
       }
       return false;
     },
+    computedEmployeeName() {
+      return this.employee.terminationId
+        ? `${this.employee.firstName} ${this.employee.lastName} 
+            (${this.$t('maintenance.past_employee')})`
+        : `${this.employee.firstName} ${this.employee.lastName}`;
+    },
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue:192]: use `translateEmployee` composable

File: src/client/src/orangehrmClaimPlugin/pages/employeeClaims/EmployeeClaims.vue [hunk 4]
```diff
@@ -212,7 +212,10 @@ export default {
             item.status.charAt(0).toUpperCase() +
             item.status.slice(1).toLowerCase(),
           submittedDate: item.submittedDate,
-          amount: parseFloat(item.amount).toFixed(2),
+          amount: Number(item.amount).toLocaleString('en-US', {
+            minimumFractionDigits: 2,
+            maximumFractionDigits: 2,
+          }),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/employeeClaims/EmployeeClaims.vue:218]: *

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 5]
```diff
@@ -311,7 +339,9 @@ public function getAll(): EndpointResult
         $count = $this->getClaimService()->getClaimDao()
             ->getClaimRequestCount($employeeClaimRequestSearchFilterParams);
 
-        return $this->getEndPointCollectionResult($claimRequests, $count);
+        $model = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_MODEL);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:342]:
```suggestion
        $model = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_MODEL, self::MODEL_DEFAULT);
```

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 6]
```diff
@@ -325,10 +355,21 @@ protected function getClaimRequestSearchFilterParams(): ClaimRequestSearchFilter
     /**
      * @param array $claimRequests
      * @param int $count
+     * @param string|null $model
      * @return EndpointCollectionResult
      */
-    protected function getEndPointCollectionResult(array $claimRequests, int $count): EndpointCollectionResult
-    {
+    protected function getEndPointCollectionResult(
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:361]:
```php
    protected function getEndPointCollectionResult(
        array $claimRequests,
        int $count,
        string $model
    ): EndpointCollectionResult {
        return new EndpointCollectionResult(
            self::MODEL_MAP[$model],
            $claimRequests,
            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
        );
}

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 7]
```diff
@@ -347,7 +388,9 @@ protected function setEmpNumbers(ClaimRequestSearchFilterParams $claimRequestSea
         );
 
         if (!is_null($empNumber)) {
-            if (!$this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber) instanceof Employee) {
+            if (!$this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber(
+                $empNumber
+            ) instanceof Employee) {
                 throw $this->getRecordNotFoundException();
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:394]: @ManulMax do we need to handle this as 404 as this is `getAll`. check similar places

File: src/plugins/orangehrmClaimPlugin/Api/MyClaimRequestAPI.php [hunk 8]
```diff
@@ -214,8 +241,18 @@ protected function getClaimRequestSearchFilterParams(): ClaimRequestSearchFilter
...
      * @inheritDoc
      */
-    protected function getEndPointCollectionResult(array $claimRequests, int $count): EndpointCollectionResult
-    {
+    protected function getEndPointCollectionResult(
+        array $claimRequests,
+        int $count,
+        ?string $model
+    ): EndpointCollectionResult {
+        if ($model === EmployeeClaimRequestAPI::MODEL_SUMMARY) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/MyClaimRequestAPI.php:249]: *

File: src/plugins/orangehrmClaimPlugin/Controller/AssignClaimRequestController.php [hunk 9]
```diff
@@ -45,19 +45,14 @@ public function preRender(Request $request): void
...
-            $claimRequest->getEmployee()->getFirstName() . ' ' . $claimRequest->getEmployee()->getLastName() . ' (Past Employee)'
-            : $claimRequest->getEmployee()->getFirstName() . ' ' . $claimRequest->getEmployee()->getLastName();
 
         if ($this->getUserRoleManagerHelper()->isSelfByEmpNumber($claimRequest->getEmployee()->getEmpNumber())) {
-            $component = new Component('submit-claim');
-        } else {
-            $component = new Component('assign-claim');
-            $component->addProp(new Prop('emp-number', Prop::TYPE_NUMBER, $empNumber));
-            $component->addProp(new Prop('emp-name', Prop::TYPE_STRING, $empName));
+            $this->setResponse($this->redirect("claim/submitClaim/id/{$id}"));
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Controller/AssignClaimRequestController.php:50]:
```suggestion
            $this->setResponse($this->redirect("claim/submitClaim/id/$id"));
```
:thinking:

File: src/plugins/orangehrmClaimPlugin/Dto/ClaimRequestSearchFilterParams.php [hunk 10]
```diff
@@ -144,6 +144,9 @@ public function getToDate(): ?DateTime
      */
     public function setToDate(?DateTime $toDate): void
     {
+        if ($toDate !== null) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Dto/ClaimRequestSearchFilterParams.php:147]:
```suggestion
        if ($toDate instanceof DateTime) {
```
I prefer this. but this is fine as param type already defined

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 11]
```diff
@@ -327,8 +355,18 @@ protected function getClaimRequestSearchFilterParams(): ClaimRequestSearchFilter
...
+        int $count,
+        ClaimRequestSearchFilterParams $claimRequestSearchFilterParams
+    ): EndpointCollectionResult {
+        if ($claimRequestSearchFilterParams->getModel() === self::MODEL_SUMMARY) {
+            return new EndpointCollectionResult(
+                ClaimRequestSummaryModel::class,
+                $claimRequests,
+                new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+            );
+        }
```
[the remark this answers is not in the document]
Comment (RajithaKumara) [reply src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php]: https://github.com/orangehrm/orangehrm/pull/1692#discussion_r1205011218

File: src/plugins/orangehrmClaimPlugin/Api/MyClaimRequestAPI.php [hunk 12]
```diff
@@ -214,8 +241,18 @@ protected function getClaimRequestSearchFilterParams(): ClaimRequestSearchFilter
...
+        int $count,
+        ClaimRequestSearchFilterParams $claimRequestSearchFilterParams
+    ): EndpointCollectionResult {
+        if ($claimRequestSearchFilterParams->getModel() === MyClaimRequestAPI::MODEL_SUMMARY) {
+            return new EndpointCollectionResult(
+                ClaimRequestSummaryModel::class,
+                $claimRequests,
+                new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+            );
+        }
```
[the remark this answers is not in the document]
Comment (RajithaKumara) [reply src/plugins/orangehrmClaimPlugin/Api/MyClaimRequestAPI.php]: https://github.com/orangehrm/orangehrm/pull/1692#discussion_r1205011218

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php (hunk 7 above)
Comment (ManulMax) [reply src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:394]: No, We handled this in getOnes.

File: src/client/src/orangehrmClaimPlugin/pages/SaveClaimEvent.vue (hunk 2 above)
Comment (ManulMax) [reply src/client/src/orangehrmClaimPlugin/pages/SaveClaimEvent.vue:122]: We have to validate in BE. @DulsaraNethmin you can check src/plugins/orangehrmAdminPlugin/Api/UserAPI.php
`new Rule(Rules::ENTITY_UNIQUE_PROPERTY, $uniquePropertyParams)`

Comment (DulsaraNethmin) [reply src/client/src/orangehrmClaimPlugin/pages/SaveClaimEvent.vue:122]: BE validation implemented

**Pull request merged**